h/form.py

```diff
@@ -15,16 +20,29 @@
 )
 
 
-def init():
-    from deform import Form
-    from deform_jinja2 import jinja2_renderer_factory
-    from deform_jinja2.translator import PyramidTranslator
+# We supply our own Jinja2 renderer factory because deform_jinja2's one doesn't
+# support autoescaping.
+def jinja2_renderer_factory(config):
```

Comment:
The `jinja2_renderer_factory` in `deform_jinja2` is a class, instances of which are callable. This is a nice pattern for factories which return callables, and avoids some of the ugliness of closures and the stuff you've had to do here with `environment = []`.

I'd suggest adopting the same pattern, and passing in the parent environment and a list of search paths rather than the whole config and relying on a global. This will make it much easier to test the behaviour of the code.

For example:

```python
class RendererFactory(object):
    def __init__(self, parent_env, search_paths=()):
        ...
        self.env = ...

    def __call__(self, template_name, **kwargs):
        template = self.env.get_template(...)
        ...
```